pkg/operation/botanist/controlplane.go

```diff
@@ -152,6 +152,24 @@ func (b *Botanist) DeployVerticalPodAutoscaler(ctx context.Context) error {
 	)
 }
 
+func (b *Botanist) determineControllerReplicas(ctx context.Context, deploymentName string, defaultReplicas int32) (int32, error) {
```

Comment:
Hint for future work: Also handle `b.Shoot.GetInfo().Status.LastOperation.Type == gardencorev1beta1.LastOperationTypeDelete` so that we don't have to set the replicas in the delete flow separately.

Reply:
Good point. Let's first integrate some more integration tests so that we can ensure functionality without manual effort, but I like the suggestion! :)